internal/check/engine_test.go

```diff
@@ -20,13 +20,103 @@ import (
 	"github.com/ory/keto/internal/driver"
 )
 
-func newDepsProvider(t *testing.T, namespaces []*namespace.Namespace, pageOpts ...x.PaginationOptionSetter) *relationtuple.ManagerWrapper {
+type configProvider = config.Provider
+type loggerProvider = x.LoggerProvider
+// deps is defined to capture engine dependencies in a single struct
+type deps struct {
+	*relationtuple.ManagerWrapper // managerProvider
+	configProvider
+	loggerProvider
+}
```

Comment:
Ah, now I get what you mean. I think the previous approach was fine as well, as the manager wrapper type is only a helper for testing, but this is of course the best way to do it :+1: